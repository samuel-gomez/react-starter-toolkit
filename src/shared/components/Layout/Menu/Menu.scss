@import '@axa-fr/react-toolkit-core/dist/assets/scss/core.scss';
@import 'shared/scss/variables.scss';

.af-title-bar__mobile-menu {
  display: none;
  background-color: $color-axa;
}

@include media-breakpoint-down(sm) {
  .af-title-bar__mobile-menu {
    position: fixed;
    display: block;
    top: calc(max(var(--titlebarHeight), var(--fullheaderHeight)) - (var(--titlebarHeight) / 2) - 1rem);
    right: 0;
    z-index: 12;
    width: 2rem;
    height: 2rem;
  }
}

.af-nav.af-drawer.left {
  z-index: 17;
}

.af-nav__list--children {
  .af-nav__item--separator {
    text-transform: uppercase;
    background: $color-gray-2;
    &:hover {
      .af-nav__link {
        color: $color-texte;
        background: $color-gray-2;
      }
    }
  }
}

.af-menu {
  padding: 0;
  margin: 0;
  &--fullscreen {
    .container {
      width: 100%;
    }
  }
}
