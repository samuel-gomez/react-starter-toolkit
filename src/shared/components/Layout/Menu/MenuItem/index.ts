export { default, createMenu } from './MenuItem.container';
export { MenuItem } from './MenuItem';
export type { TActionElt, TMenuItem, TMenuItemEnhanced, TMenuItemBase } from './MenuItem.d';
