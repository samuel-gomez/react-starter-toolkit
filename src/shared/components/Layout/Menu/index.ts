export { default } from './Menu.container';
