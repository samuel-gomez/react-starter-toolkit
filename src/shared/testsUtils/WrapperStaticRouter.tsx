/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
