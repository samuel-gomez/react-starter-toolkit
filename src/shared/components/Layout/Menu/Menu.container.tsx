import { useState, useCallback, Dispatch, SetStateAction, useMemo } from 'react';
import { usePathname } from 'next/navigation';
import Action from '@axa-fr/react-toolkit-action';
import { ToggleButton } from '@axa-fr/react-toolkit-layout-header';
import { Menu, TMenu } from './Menu';
import MENU_ITEMS, { CLASS_BODY_MENU_OPEN } from './constants';
import { TMenuItemBase } from './MenuItem';

const isNotNull = (item: unknown) => item !== null;

type TsetPositionInit = {
  menuItems: TMenuItemBase[];
  pathname: string | null;
  isChildren?: boolean;
};
export const setPositionInit = ({ menuItems, pathname, isChildren }: TsetPositionInit) => {
  const listItemsPosition: (number | null)[] = menuItems.map((navItem, index) => {
    const subIndex: number | null | undefined = navItem.children
      ? setPositionInit({ menuItems: navItem.children, pathname, isChildren: true })
      : undefined;
    if (isChildren) {
      return pathname === navItem.url ? index : null;
    }
    return pathname === navItem.url || subIndex !== undefined ? index : null;
  });
  return listItemsPosition.find(isNotNull);
};

type TsetToggleMenu = {
  setIsMenuVisible: Dispatch<SetStateAction<boolean>>;
  isVisible: boolean;
  documentObj?: Document;
};

export const setToggleMenu = ({ setIsMenuVisible, isVisible, documentObj = document }: TsetToggleMenu) => {
  const { body } = documentObj;
  body.classList.toggle(CLASS_BODY_MENU_OPEN);
  setIsMenuVisible(!isVisible);
};

export const useMenuVisible = (initState = false) => {
  const [isVisible, setIsMenuVisible] = useState(initState);
  return { isVisible, setIsMenuVisible };
};

export const computeMenuItems = (menuItems: TMenu['menuItems'] = [], basePathParent?: string) =>
  menuItems.map(({ basePathChildren, ...item }) => {
    if (item.children) {
      const children: TMenuItemBase[] = basePathChildren ? computeMenuItems(item.children, basePathChildren) : computeMenuItems(item.children);

      return { ...item, children };
    }

    if (basePathParent && item.url) {
      return { ...item, url: `${basePathParent}/${item.url}` };
    }
    return item;
  });

type TMenuEnhanced = {
  setPositionInitFn?: typeof setPositionInit;
  setToggleMenuFn?: typeof setToggleMenu;
  usePathnameFn?: typeof usePathname;
  useMenuVisibleFn?: typeof useMenuVisible;
  computeMenuItemsFn?: typeof computeMenuItems;
  menuItems?: TMenu['menuItems'];
} & TMenu;

const MenuEnhanced = ({
  menuItems = MENU_ITEMS,
  setPositionInitFn = setPositionInit,
  setToggleMenuFn = setToggleMenu,
  usePathnameFn = usePathname,
  useMenuVisibleFn = useMenuVisible,
  computeMenuItemsFn = computeMenuItems,
  ...rest
}: TMenuEnhanced) => {
  const computedMenuItems = useMemo(() => computeMenuItemsFn(menuItems), [computeMenuItemsFn, menuItems]);
  const pathname = usePathnameFn();
  const newPathname = useMemo(() => pathname?.replace(/\/+$/, '') || pathname, [pathname]);
  const positionInit = useMemo(
    () => setPositionInitFn({ menuItems: computedMenuItems, pathname: newPathname }),
    [computedMenuItems, newPathname, setPositionInitFn],
  );
  const { isVisible, setIsMenuVisible } = useMenuVisibleFn();

  const toggleMenu = useCallback(() => setToggleMenuFn({ setIsMenuVisible, isVisible }), [isVisible, setIsMenuVisible, setToggleMenuFn]);

  return (
    <>
      <Menu {...rest} menuItems={computedMenuItems} isVisible={isVisible} onClick={toggleMenu} positionInit={positionInit || 0} />
      <ToggleButton idControl="mainmenu">
        <Action
          className="btn af-title-bar__mobile-menu af-btn--circle"
          id="opentogglemenu"
          icon="menu-hamburger"
          title="Open toggle menu"
          onClick={toggleMenu}
        />
      </ToggleButton>
    </>
  );
};

export default MenuEnhanced;
