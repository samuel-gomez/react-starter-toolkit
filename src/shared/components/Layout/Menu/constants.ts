import { ROUTE_URL_HOME as HOME } from 'components/Home/constants';
import { ROUTE_URL_MEMBERS as MEMBERS } from 'components/Demos/Members/constants';
import { ROUTE_URL_SEARCHMEMBERS as SEARCHMEMBERS } from 'components/Demos/SearchMembers/constants';
import { ROUTE_URL_MODAL_CUSTOM as MODAL_CUSTOM } from 'components/Demos/ModalCustom/constants';
import { ROUTE_URL_NOTIFICATION as NOTIFICATION } from 'components/Demos/Notification/constants';
import { ROUTE_URL_NOTFOUND as NOTFOUND } from 'components/NotFound/constants';
import { ROUTE_URL_SLASH as SLASH } from 'components/Demos/SlashDesignSystem/constants';
import { ROUTE_URL_UNAUTHORIZE as UNAUTHORIZE } from 'components/Unauthorize/constants';
import { ROUTE_URL_ACCORDION as ACCORDION } from 'components/Demos/Accordion/constants';
import { ROUTE_URL_ACTION as ACTION } from 'components/Demos/Action/constants';
import { ROUTE_URL_ALERT as ALERT } from 'components/Demos/Alert/constants';
import { ROUTE_URL_BADGE as BADGE } from 'components/Demos/Badge/constants';
import { ROUTE_URL_BUTTON as BUTTON } from 'components/Demos/Button/constants';
import { ROUTE_URL_CHECKBOX_INPUT as CHECKBOX_INPUT } from 'components/Demos/CheckboxInput/constants';
import { ROUTE_URL_DATE_INPUT as DATE_INPUT } from 'components/Demos/DateInput/constants';
import { ROUTE_URL_FILE_INPUT as FILE_INPUT } from 'components/Demos/FileInput/constants';
import { ROUTE_URL_FOOTER as FOOTER } from 'components/Demos/Footer/constants';
import { ROUTE_URL_FOOTER_CLIENT as FOOTER_CLIENT } from 'components/Demos/FooterClient/constants';
import { ROUTE_URL_HEADER as HEADER } from 'components/Demos/Header/constants';
import { ROUTE_URL_HELP as HELP } from 'components/Demos/Help/constants';
import { ROUTE_URL_INFOS as INFOS } from 'components/Demos/Infos/constants';
import { ROUTE_URL_LOADER as LOADER } from 'components/Demos/Loader/constants';
import { ROUTE_URL_MODAL as MODAL } from 'components/Demos/Modal/constants';
import { ROUTE_URL_NAVBAR as NAVBAR } from 'components/Demos/NavBar/constants';
import { ROUTE_URL_NUMBER_INPUT as NUMBER_INPUT } from 'components/Demos/NumberInput/constants';
import { ROUTE_URL_POPOVER as POPOVER } from 'components/Demos/Popover/constants';
import { ROUTE_URL_RADIO_INPUT as RADIO_INPUT } from 'components/Demos/RadioInput/constants';
import { ROUTE_URL_RESTITUTION as RESTITUTION } from 'components/Demos/Restitution/constants';
import { ROUTE_URL_SELECT_INPUT as SELECT_INPUT } from 'components/Demos/SelectInput/constants';
import { ROUTE_URL_SELECT_MULTI as SELECT_MULTI } from 'components/Demos/SelectMulti/constants';
import { ROUTE_URL_SLIDER as SLIDER } from 'components/Demos/Slider/constants';
import { ROUTE_URL_STEPPER as STEPPER } from 'components/Demos/Stepper/constants';
import { ROUTE_URL_SWITCH as SWITCH } from 'components/Demos/Switch/constants';
import { ROUTE_URL_TABLE as TABLE } from 'components/Demos/Table/constants';
import { ROUTE_URL_TABS as TABS } from 'components/Demos/Tabs/constants';
import { ROUTE_URL_TEXTAREA_INPUT as TEXTAREA_INPUT } from 'components/Demos/TextareaInput/constants';
import { ROUTE_URL_TEXT_INPUT as TEXT_INPUT } from 'components/Demos/TextInput/constants';
import { ROUTE_URL_TITLE as TITLE } from 'components/Demos/Title/constants';
import { ROUTE_URL_TITLE_BAR as TITLE_BAR } from 'components/Demos/TitleBar/constants';
import { ROUTE_URL_LAYOUT as LAYOUT } from 'components/Demos/Layout/constants';

export const CLASS_BODY_MENU_OPEN = 'af-menu-open';

const DEMOS = '/demos';

const URLS_FULL_DEMOS = {
  HOME,
  MEMBERS,
  SEARCHMEMBERS,
  MODAL_CUSTOM,
  NOTIFICATION,
};

const URLS_TOOLKIT_DEMOS = {
  DEMOS,
  ACCORDION,
  ACTION,
  ALERT,
  BADGE,
  BUTTON,
  CHECKBOX_INPUT,
  DATE_INPUT,
  FILE_INPUT,
  FOOTER,
  FOOTER_CLIENT,
  HEADER,
  HELP,
  INFOS,
  LOADER,
  MODAL,
  NAVBAR,
  NUMBER_INPUT,
  POPOVER,
  RADIO_INPUT,
  RESTITUTION,
  SELECT_INPUT,
  SELECT_MULTI,
  SLASH,
  SLIDER,
  STEPPER,
  SWITCH,
  TABLE,
  TABS,
  TEXTAREA_INPUT,
  TEXT_INPUT,
  TITLE,
  TITLE_BAR,
};

const URLS_OTHER_PAGES = {
  NOTFOUND,
  UNAUTHORIZE,
  LAYOUT,
};

const ROUTE_URL = {
  ...URLS_FULL_DEMOS,
  ...URLS_TOOLKIT_DEMOS,
  ...URLS_OTHER_PAGES,
};

const MENU_ITEMS = [
  {
    label: 'Accueil',
    url: ROUTE_URL.HOME,
  },
  {
    label: 'DÃ©mos',
    url: ROUTE_URL.DEMOS,
    basePathChildren: ROUTE_URL.DEMOS,
    children: [
      {
        label: 'Membres',
        url: ROUTE_URL.MEMBERS,
      },
      {
        label: 'Rechercher',
        url: ROUTE_URL.SEARCHMEMBERS,
      },
      {
        label: 'Modal',
        url: ROUTE_URL.MODAL_CUSTOM,
      },
      {
        label: 'Notification',
        url: ROUTE_URL.NOTIFICATION,
      },
    ],
  },
  {
    label: 'Toolkit components',
    url: ROUTE_URL.DEMOS,
    basePathChildren: ROUTE_URL.DEMOS,
    children: [
      {
        label: 'Accordion',
        url: ROUTE_URL.ACCORDION,
      },
      {
        label: 'Action',
        url: ROUTE_URL.ACTION,
      },
      {
        label: 'Alert',
        url: ROUTE_URL.ALERT,
      },
      {
        label: 'Badge',
        url: ROUTE_URL.BADGE,
      },
      {
        label: 'Button',
        url: ROUTE_URL.BUTTON,
      },
      {
        label: 'Help',
        url: ROUTE_URL.HELP,
      },
      {
        label: 'Loader',
        url: ROUTE_URL.LOADER,
      },
      {
        label: 'Modal',
        url: ROUTE_URL.MODAL,
      },
      {
        label: 'Popover',
        url: ROUTE_URL.POPOVER,
      },
      {
        label: 'Restitution',
        url: ROUTE_URL.RESTITUTION,
      },
      {
        label: 'Table',
        url: ROUTE_URL.TABLE,
      },
      {
        label: 'Tabs',
        url: ROUTE_URL.TABS,
      },

      {
        label: 'Title',
        url: ROUTE_URL.TITLE,
      },
    ],
  },
  {
    label: 'Toolkit structure',
    url: ROUTE_URL.DEMOS,
    basePathChildren: ROUTE_URL.DEMOS,
    children: [
      {
        label: 'Footer',
        url: ROUTE_URL.FOOTER,
      },
      {
        label: 'Footer Client',
        url: ROUTE_URL.FOOTER_CLIENT,
      },
      {
        label: 'Header',
        url: ROUTE_URL.HEADER,
      },
      {
        label: 'Infos',
        url: ROUTE_URL.INFOS,
      },
      {
        label: 'NavBar',
        url: ROUTE_URL.NAVBAR,
      },
      {
        label: 'TitleBar',
        url: ROUTE_URL.TITLE_BAR,
      },
    ],
  },
  {
    label: 'Toolkit form components',
    url: ROUTE_URL.DEMOS,
    basePathChildren: ROUTE_URL.DEMOS,
    children: [
      {
        label: 'Checkbox Input',
        url: ROUTE_URL.CHECKBOX_INPUT,
      },
      {
        label: 'Date Input',
        url: ROUTE_URL.DATE_INPUT,
      },
      {
        label: 'File',
        url: ROUTE_URL.FILE_INPUT,
      },
      {
        label: 'Number Input',
        url: ROUTE_URL.NUMBER_INPUT,
      },
      {
        label: 'Radio Input',
        url: ROUTE_URL.RADIO_INPUT,
      },
      {
        label: 'Select Input',
        url: ROUTE_URL.SELECT_INPUT,
      },
      {
        label: 'Select Multi',
        url: ROUTE_URL.SELECT_MULTI,
      },
      {
        label: 'Slider',
        url: ROUTE_URL.SLIDER,
      },
      {
        label: 'Stepper',
        url: ROUTE_URL.STEPPER,
      },
      {
        label: 'Switch',
        url: ROUTE_URL.SWITCH,
      },
      {
        label: 'Textarea Input',
        url: ROUTE_URL.TEXTAREA_INPUT,
      },
      {
        label: 'Text Input',
        url: ROUTE_URL.TEXT_INPUT,
      },
    ],
  },
  {
    label: 'Pages',
    children: [
      {
        label: 'Not found',
        url: ROUTE_URL.NOTFOUND,
      },
      {
        label: 'Forbidden',
        url: ROUTE_URL.UNAUTHORIZE,
      },
    ],
  },
  {
    label: 'Layout',
    url: `${DEMOS}/${ROUTE_URL.LAYOUT}`,
  },
];

export default MENU_ITEMS;
