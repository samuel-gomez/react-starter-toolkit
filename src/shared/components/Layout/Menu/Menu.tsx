import { NavBar } from '@axa-fr/react-toolkit-layout-header';
import { createMenu, TMenuItemBase } from './MenuItem';
import './Menu.scss';

export type TMenu = {
  menuItems?: TMenuItemBase[];
  createMenuFn?: typeof createMenu;
  isVisible: boolean;
  onClick: () => void;
  positionInit?: number;
  fullScreen?: boolean;
};

export const Menu = ({ fullScreen, menuItems, createMenuFn = createMenu, onClick, ...navBarProps }: TMenu) => (
  <section className={`af-menu${fullScreen ? ' af-menu--fullscreen' : ''}`}>
    <NavBar {...navBarProps} onClick={onClick}>
      {menuItems && createMenuFn(menuItems, onClick)}
    </NavBar>
  </section>
);
