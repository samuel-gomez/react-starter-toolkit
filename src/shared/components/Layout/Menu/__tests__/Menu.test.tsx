import { renderWithWrapperStaticRouter } from 'shared/testsUtils';
import { Menu } from '../Menu';
import MENU_ITEMS from '../constants';

describe('<Menu />', () => {
  it('Render <Menu />', () => {
    const onClickFn = jest.fn();
    const { asFragment } = renderWithWrapperStaticRouter(<Menu menuItems={MENU_ITEMS} isVisible onClick={onClickFn} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
