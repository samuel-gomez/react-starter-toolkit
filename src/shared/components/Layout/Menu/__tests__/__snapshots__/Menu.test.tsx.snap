// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Render <Menu /> 1`] = `
<DocumentFragment>
  <section
    class="af-menu"
  >
    <div
      class="af-nav-container"
    >
      <button
        aria-controls="mainmenu"
        aria-haspopup="true"
        aria-label="Open Menu"
        class="mask fade show"
        id="togglemenu"
        type="button"
      />
      <nav
        aria-label="Menu principal"
        class="af-nav af-drawer left true"
        role="navigation"
      >
        <div
          class="container af-nav__wrapper"
        >
          <div
            class="af-nav__aside-header"
          >
            <h4
              class="af-nav__aside-title"
            >
              Menu
            </h4>
            <button
              aria-controls="mainmenu"
              aria-haspopup="true"
              aria-label="Close Menu"
              class="af-nav__aside-close"
              id="togglemenu"
              type="button"
            >
              ×
            </button>
          </div>
          <ul
            aria-labelledby="togglemenu"
            class="af-nav__list "
            id="mainmenu"
            role="menubar"
          >
            <li
              class="af-nav__item af-nav__item--active "
              role="none"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Accueil"
                class="af-nav__link"
                href="/"
                index="0"
                role="menuitem"
                tabindex="0"
              >
                Accueil
              </a>
            </li>
            <li
              class="af-nav__item af-nav__item--haschild "
              role="none"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Démos"
                basepathchildren="/demos"
                class="af-nav__link"
                href="/demos"
                index="1"
                role="menuitem"
                tabindex="-1"
              >
                Démos
              </a>
              <i
                class="glyphicon glyphicon-arrow-xs-down"
              />
              <ul
                aria-label=""
                class="af-nav__list af-nav__list--children"
                role="menu"
              >
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Membres"
                    class="af-nav__link"
                    href="/members"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Membres
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Rechercher"
                    class="af-nav__link"
                    href="/searchMembers"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Rechercher
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Modal"
                    class="af-nav__link"
                    href="/modalCustom"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Modal
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Notification"
                    class="af-nav__link"
                    href="/notification"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Notification
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="af-nav__item af-nav__item--haschild "
              role="none"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Toolkit components"
                basepathchildren="/demos"
                class="af-nav__link"
                href="/demos"
                index="2"
                role="menuitem"
                tabindex="-1"
              >
                Toolkit components
              </a>
              <i
                class="glyphicon glyphicon-arrow-xs-down"
              />
              <ul
                aria-label=""
                class="af-nav__list af-nav__list--children"
                role="menu"
              >
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Accordion"
                    class="af-nav__link"
                    href="/accordion"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Accordion
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Action"
                    class="af-nav__link"
                    href="/action"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Action
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Alert"
                    class="af-nav__link"
                    href="/alert"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Alert
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Badge"
                    class="af-nav__link"
                    href="/badge"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Badge
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Button"
                    class="af-nav__link"
                    href="/button"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Button
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Help"
                    class="af-nav__link"
                    href="/help"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Help
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Loader"
                    class="af-nav__link"
                    href="/loader"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Loader
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Modal"
                    class="af-nav__link"
                    href="/modal"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Modal
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Popover"
                    class="af-nav__link"
                    href="/popover"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Popover
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Restitution"
                    class="af-nav__link"
                    href="/restitution"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Restitution
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Table"
                    class="af-nav__link"
                    href="/table"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Table
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Tabs"
                    class="af-nav__link"
                    href="/tabs"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Tabs
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Title"
                    class="af-nav__link"
                    href="/title"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Title
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="af-nav__item af-nav__item--haschild "
              role="none"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Toolkit structure"
                basepathchildren="/demos"
                class="af-nav__link"
                href="/demos"
                index="3"
                role="menuitem"
                tabindex="-1"
              >
                Toolkit structure
              </a>
              <i
                class="glyphicon glyphicon-arrow-xs-down"
              />
              <ul
                aria-label=""
                class="af-nav__list af-nav__list--children"
                role="menu"
              >
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Footer"
                    class="af-nav__link"
                    href="/footer"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Footer
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Footer Client"
                    class="af-nav__link"
                    href="/footerClient"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Footer Client
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Header"
                    class="af-nav__link"
                    href="/header"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Header
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Infos"
                    class="af-nav__link"
                    href="/infos"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Infos
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="NavBar"
                    class="af-nav__link"
                    href="/navbar"
                    role="menuitem"
                    tabindex="-1"
                  >
                    NavBar
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="TitleBar"
                    class="af-nav__link"
                    href="/titlebar"
                    role="menuitem"
                    tabindex="-1"
                  >
                    TitleBar
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="af-nav__item af-nav__item--haschild "
              role="none"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Toolkit form components"
                basepathchildren="/demos"
                class="af-nav__link"
                href="/demos"
                index="4"
                role="menuitem"
                tabindex="-1"
              >
                Toolkit form components
              </a>
              <i
                class="glyphicon glyphicon-arrow-xs-down"
              />
              <ul
                aria-label=""
                class="af-nav__list af-nav__list--children"
                role="menu"
              >
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Checkbox Input"
                    class="af-nav__link"
                    href="/checkboxInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Checkbox Input
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Date Input"
                    class="af-nav__link"
                    href="/dateInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Date Input
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="File"
                    class="af-nav__link"
                    href="/fileInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    File
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Number Input"
                    class="af-nav__link"
                    href="/numberInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Number Input
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Radio Input"
                    class="af-nav__link"
                    href="/radioInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Radio Input
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Select Input"
                    class="af-nav__link"
                    href="/selectInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Select Input
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Select Multi"
                    class="af-nav__link"
                    href="/selectMulti"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Select Multi
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Slider"
                    class="af-nav__link"
                    href="/slider"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Slider
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Stepper"
                    class="af-nav__link"
                    href="/stepper"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Stepper
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Switch"
                    class="af-nav__link"
                    href="/switch"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Switch
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Textarea Input"
                    class="af-nav__link"
                    href="/textareaInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Textarea Input
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Text Input"
                    class="af-nav__link"
                    href="/textInput"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Text Input
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="af-nav__item af-nav__item--haschild "
              role="none"
            >
              <span
                aria-expanded="false"
                class="af-nav__link"
                index="5"
                role="menuitem"
                tabindex="-1"
              >
                Pages
              </span>
              <i
                class="glyphicon glyphicon-arrow-xs-down"
              />
              <ul
                aria-label=""
                class="af-nav__list af-nav__list--children"
                role="menu"
              >
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Not found"
                    class="af-nav__link"
                    href="/notFound"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Not found
                  </a>
                </li>
                <li
                  class="af-nav__item "
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Forbidden"
                    class="af-nav__link"
                    href="/unauthorize"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Forbidden
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="af-nav__item "
              role="none"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Layout"
                class="af-nav__link"
                href="/demos/layout"
                index="6"
                role="menuitem"
                tabindex="-1"
              >
                Layout
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </section>
</DocumentFragment>
`;
